OrbitQt/AccessibilityAdapter.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "AccessibilityAdapter.h"
+
+#include <QWidget>
+
+namespace orbit_qt {
+
+absl::flat_hash_map<const orbit_gl::GlAccessibleInterface*, QAccessibleInterface*>
+    A11yAdapter::interface_map_;
+bool A11yAdapter::initialized_ = false;
+
+//================ A11yAdapter ==================
+
+void A11yAdapter::Init() {
+  orbit_gl::GlAccessibleInterfaceRegistry::Get().OnUnregistered(
+      [](orbit_gl::GlAccessibleInterface* iface) {
+        if (interface_map_.contains(iface)) {
+          auto adapter = dynamic_cast<A11yAdapter*>(interface_map_[iface]);
+          if (adapter != nullptr) {
+            delete adapter;
+          }
+          interface_map_.erase(iface);
+        }
+      });
+  initialized_ = true;
+}
+
+QAccessibleInterface* A11yAdapter::GetOrCreateAdapter(
+    const orbit_gl::GlAccessibleInterface* iface) {
+  if (!initialized_) {
```

Comment:
I assume we don't have synchronization problems here, but this use of an effectively global variable does look worrying from that point of view.

Reply:
If we do have synchronization problems, then this whole method is problematic. At this point we shouldn't, but depending on how we use this in the future we might... but in this case, there's a lot more that needs to be looked at I think.

I wouldn't call this variable "global" since it's a private static member, but I can see the point. The reason here is that I wanted to avoid forcing an explicit call to something like `AccessibilityAdapter::Init()` which could be forgotten and would then result in leaking memory for created adapters. On the other hand, I could have added a seperate singleton class to provide the static functionality of `AccessibilityAdapter`, similar to `GlAccessibleInterfaceRegistry`, but would either make the `GetOrCreateAdapter` calls more ugly, or more costly, or both...

Reply:
Basically `std::call_once` does the same as I'm doing now, just in a concise and thread-safe manner, doesn't it? But this looks nicer, I changed it to use call_once with a locally defined static flag, this keeps it nicely hidden from the outside. No need to change the name of Init(), because it doesn't deal with the flag itself anymore.

Reply:
> Basically std::call_once does the same as I'm doing now, just in a concise and thread-safe manner, doesn't it?

Exactly, but that's why that's better.